### PR TITLE
WIP: Add support to quota replica sets

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Adds support to quota replica sets.

```
$ kubectl create quota my-quota --hard=replicasets=10
```

This brings replica sets into parity with existing support to quota replication controllers.

/cc @kubernetes/rh-cluster-infra @smarterclayton

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29868)

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
This is WIP until I can find time to get back and add E2E.  A follow-on will also look at adding support for deployments to be under quota as well.

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -3024,6 +3024,8 @@ const (
 	ResourceServices ResourceName = "services"
 	// ReplicationControllers, number
 	ResourceReplicationControllers ResourceName = "replicationcontrollers"
+	// ReplicaSets, number
+	ResourceReplicaSets ResourceName = "replicasets"
```
Comment (deads2k) [inline pkg/api/v1/types.go]: I prefer @mfojtik's approach of moving this to the API group from whence it came.  I'd like to establish a pattern that federators can use when adding their own group.  `groupName/resourceName` seems reasonable.

Comment (derekwaynecarr) [inline pkg/api/v1/types.go]: yeah, i was still debating the same.  I have no objection to changing it to be `groupName/resourceName`

**Comment by derekwaynecarr** [comment]
Closing in favor @mfojtik PR.

**Pull request closed without merging**